internal/cli/atlas/deployments/start.go

```diff
@@ -0,0 +1,174 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deployments
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/atlas/deployments/options"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/atlas/setup"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/log"
+	"github.com/mongodb/mongodb-atlas-cli/internal/podman"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+	"go.mongodb.org/atlas-sdk/v20230201008/admin"
+)
+
+type StartOpts struct {
+	cli.OutputOpts
+	cli.GlobalOpts
+	options.DeploymentOpts
+	store  store.ClusterStarter
+	config setup.ProfileReader
+}
+
+var startTemplate = "Starting deployment '{{.Name}}'.\n"
+
+var ErrDeploymentIsDeleting = errors.New("deployment state is DELETING")
+
+func (opts *StartOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *StartOpts) Run(ctx context.Context) error {
+	if err := opts.validateAndPrompt(ctx); err != nil {
+		return err
+	}
+
+	if strings.ToLower(opts.DeploymentType) == options.LocalCluster {
+		return opts.RunLocal(ctx)
+	}
+
+	return opts.RunAtlas()
+}
+
+func (opts *StartOpts) RunLocal(ctx context.Context) error {
+	localDeployments, err := opts.GetLocalDeploymentsWithContainers(ctx)
+	if err != nil {
+		return err
+	}
+
+	if deployment, ok := localDeployments[opts.DeploymentName]; ok {
+		if err = opts.startContainer(ctx, deployment); err != nil {
+			return err
+		}
+
+		return opts.Print(
+			admin.AdvancedClusterDescription{
+				Name: &opts.DeploymentName,
+			})
+	}
+
+	return options.ErrDeploymentNotFound
+}
+
+func (opts *StartOpts) startContainer(ctx context.Context, deployment options.Deployment) error {
+	if deployment.StateName == options.StoppedState {
+		if _, err := opts.PodmanClient.StartContainers(ctx, deployment.MongoDContainer.ID, deployment.MongoTContainer.ID); err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	if deployment.StateName == options.PausedState {
+		if _, err := opts.PodmanClient.UnpauseContainers(ctx, deployment.MongoDContainer.ID, deployment.MongoTContainer.ID); err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	if deployment.StateName == options.IdleState || deployment.StateName == options.RestartingState {
+		return nil
+	}
+
+	return ErrDeploymentIsDeleting
+}
+
+func (opts *StartOpts) RunAtlas() error {
+	r, err := opts.store.StartCluster(opts.ConfigProjectID(), opts.DeploymentName)
```

Comment:
do we need to check if user is authenticated first?

Reply:
updated